state/protocol/badger/snapshot.go

```diff
@@ -373,26 +372,22 @@ func (q *EpochQuery) Next() protocol.Epoch {
 	if err != nil {
 		return invalid.NewEpochf("could not get protocol state snapshot at block %x: %w", q.snap.blockID, err)
 	}
-	status := psSnapshot.EpochStatus()
-	phase, err := status.Phase()
-	if err != nil {
-		// critical error: malformed EpochStatus in storage
-		return invalid.NewEpochf("read malformed EpochStatus from storage: %w", err)
-	}
+	phase := psSnapshot.EpochPhase()
 	// if we are in the staking phase, the next epoch is not setup yet
 	if phase == flow.EpochPhaseStaking {
 		return invalid.NewEpoch(protocol.ErrNextEpochNotSetup)
 	}
-
 	// if we are in setup phase, return a SetupEpoch
 	nextSetup := psSnapshot.Entry().NextEpochSetup
 	if phase == flow.EpochPhaseSetup {
 		return inmem.NewSetupEpoch(nextSetup)
 	}
-
 	// if we are in committed phase, return a CommittedEpoch
 	nextCommit := psSnapshot.Entry().NextEpochCommit
```

Comment:
Currently, we are duplicating the call to `psSnapshot.Entry()` in [line 381](https://github.com/onflow/flow-go/blob/04b42dc36b5829baabf62e8201a1737afe770451/state/protocol/badger/snapshot.go#L381) and [line 386](https://github.com/onflow/flow-go/blob/04b42dc36b5829baabf62e8201a1737afe770451/state/protocol/badger/snapshot.go#L386). I feel we could just retrieve the `RichProtocolStateEntry` once (which in the current implementation would avoid one deep copy). The result could look something like this:


```golang
// if we are in setup phase, return a SetupEpoch
richPs := psSnapshot.Entry()
if phase == flow.EpochPhaseSetup {
  return inmem.NewSetupEpoch(richPs.NextEpochSetup)
}
// if we are in committed phase, return a CommittedEpoch
if phase == flow.EpochPhaseCommitted {
  return inmem.NewCommittedEpoch(richPs.NextEpochSetup, richPs.NextEpochCommit)
}
```